### PR TITLE
Do not export Guava from Objectify library definition

### LINKED ISSUE #2381: Library export conflict build error when adding Objectify and Cloud Core Client Library together
The same error as in #2369, but with Objectify and Cloud Core Client Library on App Engine. This must have been like this (not a regression).

`Invalid classpath publish/export dependency /home/chanseok/.m2/repository/com/google/guava/guava/20.0/guava-20.0.jar. The project contains another dependency with the same archive name.`

### PR DESCRIPTION
**Pull request by briandealwis**
Add integration test that verifies no build errors in a project with all available library definitions.

Fixes #2381

### CODE REVIEW

**Comment by briandealwis** [comment]
Test should fail (commit 15878f1).
Fix should cause it to pass (commit f524b0c).

**Comment by chanseokoh** [comment]
Tests are failing. There is an explicit test that verifies Guava is exported:

```
testObjectifyLibraryConfig(com.google.cloud.tools.eclipse.appengine.libraries.model.CloudLibrariesInPluginXmlTest)  Time elapsed: 0.002 sec  <<< FAILURE!
java.lang.AssertionError: Guava not exported
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.google.cloud.tools.eclipse.appengine.libraries.model.CloudLibrariesInPluginXmlTest.testObjectifyLibraryConfig(CloudLibrariesInPluginXmlTest.java:169)
```

**Comment by chanseokoh** [comment]
I'm trying to understand the consequences of not exporting it with Objectify. Clearly, `guava-20.0.jar` is not placed in `WEB-INF/lib` in the staging area:

```
└── WEB-INF
     ...
    ├── classes
    │   └── HelloAppEngine.class
    ├── lib
    │   └── objectify-5.1.21.jar
    └── web.xml
```
Whereas before this PR, the jar will be in the `lib/` directory. Can you confirm it is OK that Objectify will work without the Guava JAR?

**Comment by chanseokoh** [comment]
Yeah, I tested a [sample Objectify app](https://www.google.com/url?q=https%3A%2F%2Fgithub.com%2FGoogleCloudPlatform%2Fjava-docs-samples%2Ftree%2Fc817c925a27f3aa0365f23981a228b39135d6ff1%2Fappengine%2Fguestbook-objectify&sa=D&sntz=1&usg=AFQjCNGL6gQ2AJMpuDy0lzQEuV-P7VWFBA), and it does require exporting Guava:

```
Uncaught exception from servlet
java.lang.NoClassDefFoundError: com/google/common/base/Function
	at com.googlecode.objectify.cache.EntityMemcache.<init>(EntityMemcache.java:178)
	at com.googlecode.objectify.cache.EntityMemcache.<init>(EntityMemcache.java:169)
	at com.googlecode.objectify.ObjectifyFactory.<init>(ObjectifyFactory.java:59)
	at com.googlecode.objectify.ObjectifyService.<clinit>(ObjectifyService.java:20)
	at com.example.guestbook.OfyHelper.contextInitialized(OfyHelper.java:29)
```

I've also verified the app works when exporting Guava as currently.

**Comment by chanseokoh** [comment]
We can talk about this tomorrow before starting things.

**Comment by briandealwis** [comment]
Extracting Guava to a separate `<library>` seems to work well.  
screen shot 2017-09-13 at 12 49 10 pm

**Comment by briandealwis** [comment]
But this will break existing projects.  Ok, we'll chat tomorrow.

**Comment by briandealwis** [comment]
Abandoned as better fix is to manage libraries in a single classpath container.

**Comment by briandealwis** [comment]
Re-opening as I pulled out Guava and fixed up the tests.  Pushed up the work in case we want to go this path as a temporary measure.

File: plugins/com.google.cloud.tools.eclipse.appengine.libraries.test/src/com/google/cloud/tools/eclipse/appengine/libraries/model/CloudLibrariesInPluginXmlTest.java [hunk 1]
```diff
@@ -40,14 +42,22 @@
...
-    assertThat(CloudLibraries.getLibraries("appengine").size(), is(3));
+    assertThat(CloudLibraries.getLibraries("appengine").size(), is(4));
+
+    // There may be 2 or 4 servlet libraries depending on whether the
+    // .appengine.java.standard.java8 bundle is present
     List<Library> servletLibraries = CloudLibraries.getLibraries("servlet");
-    // todo in Eclipse this returns 4 (versions 2 and 3 of JSP and servlet libs)
-    // figure out why
-    assertEquals("wrong number of servlet libraries", 2, servletLibraries.size());
+    // split out toString() matches because of type inference errors
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.libraries.test/src/com/google/cloud/tools/eclipse/appengine/libraries/model/CloudLibrariesInPluginXmlTest.java]: This change seems worth including, irrespective of the rest.

File: plugins/com.google.cloud.tools.eclipse.appengine.libraries/plugin.properties [hunk 2]
```diff
@@ -1,6 +1,13 @@
...
 providerName=Google Inc.
-objectify.tooltip=com.googlecode.objectify, a Java Data Access API Designed for Google Cloud Datastore	
+objectify.name=Objectify
+objectify.tooltip=com.googlecode.objectify, a Java Data Access API Designed for Google Cloud Datastore
+guava.20.name=Guava 20.0
+guava.20.tooltip=Guava is a set of core libraries that includes \
+ new collection types (such as multimap and multiset), immutable collections, \
+ a graph library, functional types, an in-memory cache, \
+ and APIs/utilities for concurrency, I/O, hashing, primitives, \
+ reflection, string processing, and much more!
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.libraries/plugin.properties]: much more! --> more.

File: plugins/com.google.cloud.tools.eclipse.appengine.libraries.test/src/com/google/cloud/tools/eclipse/appengine/libraries/model/CloudLibrariesInPluginXmlTest.java (hunk 1 above)
Comment (briandealwis) [reply plugins/com.google.cloud.tools.eclipse.appengine.libraries.test/src/com/google/cloud/tools/eclipse/appengine/libraries/model/CloudLibrariesInPluginXmlTest.java]: Will do.

**Comment by briandealwis** [comment]
> If this works, I'd be happy to merge in advance of a more complete solution.

It doesn't work well as existing projects must add the Guava dependency manually.

**Pull request closed without merging**